application/src/main/java/org/togetherjava/tjbot/commands/basic/RoleSelectCommand.java

```diff
@@ -0,0 +1,265 @@
+
+    private static final String TITLE_OPTION = "title";
+    private static final String DESCRIPTION_OPTION = "description";
+
+    private static final List<OptionData> messageOptions = List.of(
+            new OptionData(OptionType.STRING, TITLE_OPTION, "The title for the message", false),
+            new OptionData(OptionType.STRING, DESCRIPTION_OPTION, "A description for the message",
+                    false));
+
+
+    /**
+     * Construct an instance
+     *
+     * @see RoleSelectCommand
+     */
+    public RoleSelectCommand() {
+        super("role-select", "Sends a message where users can select their roles",
+                SlashCommandVisibility.GUILD);
+
+        SubcommandData allRoles =
+                new SubcommandData(ALL_OPTION, "Lists all the rolls in the server for users")
+                    .addOptions(messageOptions);
+
+        SubcommandData selectRoles =
+                new SubcommandData(CHOOSE_OPTION, "Choose the roles for users to select")
+                    .addOptions(messageOptions);
+
+        getData().addSubcommands(allRoles, selectRoles);
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull SlashCommandEvent event) {
+        Member member = Objects.requireNonNull(event.getMember(), "Member is null");
+        if (!member.hasPermission(Permission.MANAGE_ROLES)) {
+            event.reply("You dont have the right permissions to use this command")
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+
+        Member selfMember = Objects.requireNonNull(event.getGuild()).getSelfMember();
+        if (!selfMember.hasPermission(Permission.MANAGE_ROLES)) {
+            event.reply("The bot needs the manage role permissions").setEphemeral(true).queue();
+            logger.warn("The bot needs the manage role permissions");
+            return;
+        }
+
+        SelectionMenu.Builder menu = SelectionMenu.create(generateComponentId(member.getId()));
+        boolean ephemeral = false;
+
+        if (Objects.equals(event.getSubcommandName(), CHOOSE_OPTION)) {
+            addMenuOptions(event, menu, "Select the roles to display", 1);
+            ephemeral = true;
+        } else {
+            addMenuOptions(event, menu, "Select your roles", 0);
+        }
+
+        // Handle Optional arguments
+        OptionMapping titleOption = event.getOption(TITLE_OPTION);
+        OptionMapping descriptionOption = event.getOption(DESCRIPTION_OPTION);
+
+        String title = handleOption(titleOption);
+        String description = handleOption(descriptionOption);
+
+        if (ephemeral) {
+            event.replyEmbeds(makeEmbed(title, description))
+                .addActionRow(menu.build())
+                .setEphemeral(true)
+                .queue();
+        } else {
+            event.getChannel()
+                .sendMessageEmbeds(makeEmbed(title, description))
+                .setActionRow(menu.build())
+                .queue();
```

Comment:
```suggestion
            event.replyEmbeds(makeEmbed(title, description))
                .addActionRow(menu.build())
                .setEphemeral(ephemeral)
                .queue();
```

Reply:
Removing this would mean if the user used `/role-select all` the message would be sent as a reply to the command which we decided wasn't the best right?

![image](https://user-images.githubusercontent.com/52386683/144504540-35315463-3299-4be2-843c-68ee95f74fd7.png)

Reply:
Could you elaborate?
```java
.setEphemeral(ephemeral)
```
I still run this method.
This is just a lot shorter and easier readable by that.

Reply:
Never mind!

I completely missed the fact that one isn't a reply, I'll resolve this